native/core/src/execution/tracing.rs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#[cfg(feature = "tracing")]
+use datafusion::common::instant::Instant;
+#[cfg(feature = "tracing")]
+use once_cell::sync::Lazy;
+
+#[cfg(feature = "tracing")]
+pub(crate) static RECORDER: Lazy<Recorder> = Lazy::new(|| Recorder::new());
+
+/// Log events using Chrome trace format JSON
+/// https://github.com/catapult-project/catapult/blob/main/tracing/README.md
+#[cfg(feature = "tracing")]
+pub struct Recorder {
+    now: Instant,
+}
+
+#[cfg(feature = "tracing")]
+impl Recorder {
+    pub fn new() -> Self {
+        // Write start of JSON array. Note that there is no requirement to write
+        // the closing ']'.
+        print!("[ ");
+        Self {
+            now: Instant::now(),
+        }
+    }
+    pub fn begin_task(&self, name: &str) {
+        self.log_event(name, "B")
+    }
+
+    pub fn end_task(&self, name: &str) {
+        self.log_event(name, "E")
+    }
+
+    pub fn log_counter(&self, name: &str, value: usize) {
```

Comment:
Can this be `u64` and so not be system dependent?  Not sure how useful it might be but some counters (say bytes read) could be `u64` but `usize` may be smaller.

Reply:
Thanks. Updated.